stattutor/stattutor.py

```diff
@@ -0,0 +1,386 @@
+"""
+This is a XBlock used to serve the CTAT based StatTutor problems originally
+implemented for OLI (http://oli.cmu.edu/).
+"""
+
+import re
+import uuid
+import base64
+import math
+import pkg_resources
+import bleach
+
+# pylint: disable=import-error
+# The xblock packages are available in the runtime environment.
+from xblock.core import XBlock
+from xblock.fields import Scope, Integer, String, Float, Boolean
+from xblock.fragment import Fragment
+# pylint: enable=import-error
+
+
+class StattutorXBlock(XBlock):
+    """
+    A XBlock providing a CTAT backed StatTutor.
+    """
+    # pylint: disable=too-many-instance-attributes
+    # All of the instance variables are required.
+
+    display_name = String(
+        help="Display name of the component",
+        default="StatTutor",
+        scope=Scope.content)  # required to prevent garbage name at the top
+
+    # **** xBlock tag variables ****
+    # The width must be at least 900 in order to accommodate some dynamically
+    # loaded images and some of the interactive elements without causing
+    # side scrolling scrollbars to appear.  They are set here instead of
+    # hard coding them into ctatxblock.html to make it easier for EdX
+    # administrators to modify them if they wish without having to scour
+    # all of the code for where they are set.
+    width = 900  # Width of the StatTutor frame.
+    height = 750  # Height of the StatTutor frame.
+
+    # **** Grading variables ****
+    # All of the variable in this section are required to get grading to work
+    # according to EdX's documentation.
+    has_score = Boolean(default=True, scope=Scope.content)
+    icon_class = String(default="problem", scope=Scope.content)
+    score = Integer(help="Current count of correctly completed student steps",
+                    scope=Scope.user_state, default=0)
+    max_problem_steps = Integer(
```

Comment:
What determines the number of problem steps? Does it vary problem to problem?
Could you have multiple instances of problem 1 with varying numbers of steps?